datastore/mongo/group.go

```diff
@@ -173,13 +180,44 @@ func (db *groupRepo) FillGroupsStatistics(ctx context.Context, groups []*datasto
 
 func (db *groupRepo) DeleteGroup(ctx context.Context, uid string) error {
 	ctx = db.setCollectionInContext(ctx)
-
-	err := db.store.DeleteByID(ctx, uid, false)
-	if err != nil {
-		return err
+	updateAsDeleted := bson.M{
+		"$set": bson.M{
+			"deleted_at":      primitive.NewDateTimeFromTime(time.Now()),
+			"document_status": datastore.DeletedDocumentStatus,
+		},
 	}
 
-	return nil
+	err := db.store.WithTransaction(ctx, func(sessCtx mongo.SessionContext) error {
+		err := db.store.DeleteByID(sessCtx, uid, false)
+		if err != nil {
+			return err
+		}
+
+		var apps []datastore.Application
+
+		ctx := context.WithValue(sessCtx, datastore.CollectionCtx, datastore.AppCollection)
+		filter := bson.M{"group_id": uid}
+		err = db.store.FindAll(ctx, filter, nil, nil, &apps)
+		if err != nil {
+			return err
+		}
+
+		for _, app := range apps {
+			err = db.deleteAppEvents(sessCtx, uid, updateAsDeleted)
+			if err != nil {
+				return err
+			}
+
+			err = db.deleteApp(sessCtx, app.UID, updateAsDeleted)
```

Comment:
Shouldn't we delete the app's subscription as well?

Reply:
Yeah, it's becoming a mess though, lol, I'll add it